src/message.rs

```diff
@@ -214,9 +214,10 @@ impl Message {
             _ => return Ok(None)
         };
 
+        let mut body = self.body.clone();
```

Comment:
If we're just going to clone the message here, why not just expect the caller to do that if it needs to hold on to the message?  That way in the "normal" case (i.e., you don't use the message object after sending it), we can avoid the copy.